### PR TITLE
feat: Use quarter to hour function

### PR DESCRIPTION
**Pull request by johevemi**
# Description
This PR will implement the use of quarter to hour function which means that periods before quarterly_resolution_transition_datetime will now have hourly resolution and those after will have quarterly

## Pull-request quality

- [x] The title adheres to [this guide](https://github.com/Mech0z/GitHubGuidelines)
- [x] Tests are written and executed locally
- [ ] Subsystem tests have been tested (by manually deploying to `dev_002`)
- [ ] Documentation has been updated
- [ ] The integration [event catalog](https://energinet.atlassian.net/wiki/spaces/D3/pages/555581556/Event+catalog) has been updated
- [ ] C4 diagrams have been updated and Team Outlaws has been informed about relevant changes

### CODE REVIEW

File: source/databricks/calculation_engine/package/calculation/energy/resolution_transition_factory.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+from package.calculation.preparation.data_structures import (
+    PreparedMeteringPointTimeSeries,
+)
+from package.calculation.preparation.data_structures.metering_point_time_series import (
+    MeteringPointTimeSeries,
+)
+from package.codelists import MeteringPointResolution
+
+
+def get_resolution(
```
Comment (JesperGomes) [inline source/databricks/calculation_engine/package/calculation/energy/resolution_transition_factory.py:28]: get_energy_result_resolution

Comment (johevemi) [reply source/databricks/calculation_engine/package/calculation/energy/resolution_transition_factory.py:28]: the function is already in energy

File: source/databricks/calculation_engine/package/calculation/output/energy_storage_model_factory.py [hunk 2]
```diff
@@ -34,10 +37,11 @@ def create(
         energy_results.df, aggregation_level, time_series_type
     )
     df = add_metadata(args, _get_column_group_for_calculation_result_id(), df)
-    # TODO JVM: Create a function to get the correct resolution when quarter_to_hour is implemented
-    df = df.withColumn(
-        EnergyResultColumnNames.resolution, f.lit(MeteringPointResolution.QUARTER.value)
+    metering_point_resolution = get_resolution(
```
Comment (JesperGomes) [inline source/databricks/calculation_engine/package/calculation/output/energy_storage_model_factory.py:40]: isn't energy_result_resolution more appropriate? The original metering points have mixed resolutions

Comment (johevemi) [reply source/databricks/calculation_engine/package/calculation/output/energy_storage_model_factory.py:40]: yeah you are right, i will change it

**Review by JesperGomes — APPROVED** [review]

**Pull request merged**